### PR TITLE
[KOGITO-5251] - Upgrade to SpringBoot 2.4.3

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

See: https://issues.redhat.com/browse/KOGITO-5251

Keycloak is only used in SB tests. So it was a good opportunity also to upgrade it. The SpringDocs is just a patch upgrade, also worth it to make the change.

Please note that from SB 2.4.x, junit 4 is deprecated and SB Test BOM does not include it anymore.

See:
https://github.com/kiegroup/kogito-apps/pull/918
https://github.com/kiegroup/kogito-examples/pull/757

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-build/kogito-ide-config)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

File: kogito-build/kogito-build-parent/pom.xml [hunk 1]
```diff
@@ -172,7 +172,7 @@
     <version.org.junit.vintage>${version.org.junit}</version.org.junit.vintage>
     <version.org.junit.platform>1.${version.org.junit.minor}</version.org.junit.platform> <!-- otherwise Quarkus brings its own, silently disabling some tests -->
     <version.org.junit.pioneer>1.0.0</version.org.junit.pioneer>
-    <version.org.keycloak>12.0.4</version.org.keycloak>
+    <version.org.keycloak>14.0.0</version.org.keycloak>
```
Comment (danielezonca) [inline kogito-build/kogito-build-parent/pom.xml:175]: Are you sure we have to bump 2 major release of keycloak to run with newer version of Spring?

Comment (ricardozanini) [reply kogito-build/kogito-build-parent/pom.xml:175]: As I said in the PR, it's just for tests. We are far behind and looking at the release notes, I haven't found anything too relevant. I'll update apps and examples this afternoon.

File: kogito-build/kogito-build-parent/pom.xml [hunk 2]
```diff
@@ -181,8 +181,8 @@
     <version.org.reactivestreams>1.0.3</version.org.reactivestreams>
     <version.org.reflections>0.9.11</version.org.reflections>
     <version.org.slf4j>1.7.30</version.org.slf4j>
-    <version.org.springframework.boot>2.3.4.RELEASE</version.org.springframework.boot>
-    <version.org.springdoc>1.5.8</version.org.springdoc>
+    <version.org.springframework.boot>2.4.3</version.org.springframework.boot>
```
Comment (tiagodolphine) [inline kogito-build/kogito-build-parent/pom.xml:184]: Just a question, why we are upgrading to 2.4.3 and not to the latest, any reason for this specific version?

Comment (ricardozanini) [reply kogito-build/kogito-build-parent/pom.xml:184]: Product support.

**Review by tiagodolphine — APPROVED** [review]

**Comment by ricardozanini** [comment]
Jenkins run examples tests

**Comment by ricardozanini** [comment]
Jenkins run examples tests

**Comment by ricardozanini** [comment]
Jenkins run apps tests

**Review by MarianMacik — APPROVED** [review]

**Review by danielezonca — APPROVED** [review]

**Comment by ricardozanini** [comment]
The third-party starters will be aligned in this JIRA: https://issues.redhat.com/browse/KOGITO-5355

We need to make sure that each starter is aligned/supported by our SB base version.

**Pull request merged**